### DISCUSSION TITLE
Get access to Mermaid's generated SVG code

### DISCUSSION THREAD

**Discussion by groucho86** [zauberzeug/nicegui#4035] [Q&A] [answered]
### Question

Wondering if some already-existing function(s) can allow me to extract the svg code. Or would we need to add custom methods within mermaid.js

**Comment by falkoschindler** [comment, marked as answer]
Hi @groucho86,

Even though there is no explicit function for doing that, you might be able to create a workaround using `ui.run_javascript`:
```py
@ui.page('/')
def index():
    m = ui.mermaid('''
    graph TD
        A --> B
        B --> C
    ''')

    async def get_svg():
        svg = await ui.run_javascript(f'getElement({m.id}).$el.querySelector("svg").outerHTML')
        print(svg)

    ui.button('Get SVG', on_click=get_svg)
```

**Reply by groucho86** [reply]
as always, thank you very much @falkoschindler
